util/log/logrus.go

```diff
@@ -2,14 +2,56 @@ package log
 
 import (
 	"fmt"
-
+	"github.com/argoproj/argo-cd/v2/common"
 	adapter "github.com/bombsimon/logrusr"
 	"github.com/go-logr/logr"
 	"github.com/sirupsen/logrus"
+	"os"
+	"strings"
+)
+
+const (
+	JsonFormat = "json"
+	TextFormat = "text"
 )
 
 func NewLogrusLogger(fieldLogger logrus.FieldLogger) logr.Logger {
 	return adapter.NewLoggerWithFormatter(fieldLogger, func(val interface{}) string {
 		return fmt.Sprintf("%v", val)
 	})
 }
+
+// NewWithCurrentConfig create logrus logger by using current configuration
+func NewWithCurrentConfig() *logrus.Logger {
+	l := logrus.New()
+	l.SetFormatter(CreateFormatter(os.Getenv(common.EnvLogFormat)))
+	l.SetLevel(createLogLevel())
+	return l
+}
+
+// CreateFormatter create logrus formatter by string
+func CreateFormatter(logFormat string) logrus.Formatter {
+	var formatType logrus.Formatter
+	switch strings.ToLower(logFormat) {
+	case JsonFormat:
+		formatType = &logrus.JSONFormatter{}
+	case TextFormat:
+		if os.Getenv("FORCE_LOG_COLORS") == "1" {
+			formatType = &logrus.TextFormatter{ForceColors: true}
+		} else {
+			formatType = &logrus.TextFormatter{}
+		}
+	default:
+		formatType = &logrus.TextFormatter{}
+	}
+
+	return formatType
+}
+
+func createLogLevel() logrus.Level {
+	level, err := logrus.ParseLevel(os.Getenv(common.EnvLogLevel))
+	if err != nil {
+		level = logrus.InfoLevel
+	}
+	return level
+}
```

Comment:
Validation for `EnvLogLevel` is done when it is set on `SetLogLevel` method, `util/cli/cli.go`.
So I think I do not need to raise error, like `errors.CheckError(err)` (maybe error does not happen here) ... 🤔

Reply:
agree